go.mod

```diff
@@ -0,0 +1,24 @@
+module terraform-provider-zerotier
+
+go 1.12
+
+require (
+	github.com/apparentlymart/go-cidr v1.0.0 // indirect
```

Comment:
Forgive me, I'm new to go mod, but why are all these 'indirect' dependencies also locked down? Is it just because they were in the dep lockfile? Do you need this and is it necessary? I have removed all the indirects locally and it builds just fine.

Reply:
(Although it does pull a few newer versions of these packages and their transitive deps before doing so.)

Reply:
Those dependencies where introduced by `go mod tidy`. I've added only the single dependency on `terraform`, and the `// indirect` dependencies are put in place whenever I run `go mod tidy`.

Running the command was documented on the transition guide, and these lines seems to be expected to exist. Searching quickly, there is some confusing regarding this behavior on the internet. What I got was the following:

> `go mod tidy` ensures your current go.mod reflects all possible build tags/OS/architecture combinations (as described [here](https://github.com/golang/go/issues/25971#issuecomment-399091682)).
> 
> In contrast, other commands like `go build` and `go test` only update `go.mod` based on the current build invocation's tags/OS/architecture.

https://github.com/golang/go/issues/27428#issuecomment-417934130

Reply:
What I understood is that without the transitive dependencies, it might not have all the dependencies declared for my OS when building, so `go mod tidy` includes all necessary regardless of the OS that run `go mod`

Reply:
Maybe it would be nice to include `go mod tidy` on `make dev`? What do you think?

Reply:
Ok. I also had to upgrade hashicorp/terraform to 0.11.13 to match my local install (it segfaulted otherwise), and apparently terraform now [uses go mod as well](https://github.com/hashicorp/terraform/blob/master/go.mod), so these lines no longer get inserted by `go mod tidy`.

Reply:
`go mod tidy` is already in `make dev`.

Reply:
There aren't any issues with upgrading the terraform dep are there? I don't think they introduce any breaking changes to the plugin interface that won't work for say a terraform 0.11.7 binary. I think that segfault in the opposite direction was a bug.

Reply:
I don't think they've broke the integration, but I could give it a try, given I'm still on an older installation.

Reply:
I'm running `Terraform v0.11.11` and compiling with `v0.11.13` works locally.

Reply:
Good. I thought so.